src/main/java/io/pakland/mdas/githubstats/domain/Commit.java

```diff
@@ -16,30 +18,20 @@ public class Commit {
 
     @Id
     @Column(updatable = false, nullable = false)
-    private Integer id;
+    @JsonProperty("sha")
+    private String sha;
+
+    @Column(name = "date")
+    private Date date;
 
     @ManyToOne(fetch = FetchType.LAZY)
     private User user;
 
     @ManyToOne(fetch = FetchType.LAZY)
     private PullRequest pullRequest;
 
-    private int additions;
-
-    private int deletions;
-
-    private Instant date;
-
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (!(o instanceof Commit)) return false;
-        return id != null && id.equals(((Commit) o).getId());
+    @JsonProperty("commit")
+    private void unpackDateFromNestedObject(Map<String, Object> commit) {
+        //        this.date = Date.from(Instant.parse(commiter.get("date").toString()));
```

Comment:
This is still in WIP. I'm investigating how to map deep nested properties. (it's two levels deep)

Reply:
Have you tried this? https://www.baeldung.com/jackson-nested-values

I understand that all objects are converted into maps, and that it is recursively executed. Therefore, an object within a property of another object, would be a map inside a map.

Reply:
Done, working! ✅